### DISCUSSION TITLE
/posts/yabai-setup

### DISCUSSION THREAD

**Discussion by giscus** [josean-dev/sveltekit-blog#4] [General]
# /posts/yabai-setup

Use this guide along with my youtube video to setup and use the yabai tiling window manager on your Mac. Yabai has been great for leveling up my window management workflow.

https://www.josean.com/posts/yabai-setup

**Comment by lvillalobos-cr** [comment]
# Amazing Content! 

Your video and guide definitely helped me level up my coding workspaces.

**Comment by ryukato** [comment]
`brew services start yabai` does not work on my machine, so I ran `yabai --start-service` instead. :)

**Reply by josean-dev** [reply]
Just fixed this in the blog post!

**Comment by fbruggem** [comment]
The code block "skdh --start-service" is misstyped. It should be SKHD not SKDH. 
Nice tutorial otherwise keep up the good work!

**Reply by alecthegeek** [reply]
Ah -- thanks for that :-). Wondered why it did not work

**Comment by sethhorsley** [comment]
Great videos! I was curious what you use to switch applications or spaces? I programmed `alt/option + 1` and `alt/option + 2` .. to go to that number space or create it. Just wondering what your workflow is.

Also noticed your alt tab on your keyboard is on the far bottom right is that an easy spot to use or do you not use it as much?

**Comment by jokerinya** [comment]
Thank you for this great content. 🙏🏼
There is a small typo in the code snippet that you have provided. 
The `skhd` service starter command should be `skhd --start-service` not `skdh --start-service` 🙂

**Comment by ghost** [comment]
Great video and blog post, helped me streamline my mac workflow better.

**Comment by meherjammula** [comment]
Thank you This is veryy helpful

**Comment by Fenfox13** [comment]
Everything works except the skhd keyboard shortcuts. I use textedit and just saved the file and restarted skhd but it doesnt work. Directory seems to be in the right place and all. any ideas?

**Reply by Fenfox13** [reply]
another note - whatever hotkey I do just does an action in the selected app instead of skhd, like making special characters in terminal or selecting different categories in mail

**Reply by patrolek100** [reply]
I had the same issue. In my case the problem was that I am not using the default English keyboard, so stuff like ```alt - l ```makes special characters such as ł. My workaround was to add extra keys to keyboard shortcuts, now everything works. Try with each chunk of code, step by step.
To change window focus within space he uses
```
alt - j : yabai -m window --focus south
alt - k : yabai -m window --focus north
alt - h : yabai -m window --focus west
alt - l : yabai -m window --focus east
```
which for me triggered those special characters and broke the code. I have added ```cmd``` key to all of those shortcuts, which looks like this:
``` 
cmd + alt - j : yabai -m window --focus south
cmd + alt - k : yabai -m window --focus north
cmd + alt - h : yabai -m window --focus west
cmd + alt - l : yabai -m window --focus east
```
save the config file, restart skhd using ```skhd --restart-service``` and check if it works.

For more advanced keyboard shortcuts I'm using ```cmd + alt + shift``` and to go even more granular ```cmd + alt + shift + ctrl```.

Hope it helps : )

**Reply by paurana** [reply]
I was facing the same issue. `nohup skhd &` fixed it for me

**Reply by padi** [reply]
If your shortcuts work on other apps except for iTerm2, here's the fix:

Took me more than an hour to figure this out for `alt - j`, `alt - k`, `alt - j`, `alt - l` shortcuts and everything else on skhdrc.

Disable iterm2 > "Secure Keyboard Entry" and shortcuts that work on other apps should work on iTerm2. 

Some people report that Preferences > Profiles > `<Your iterm2 profile name>` > Keys > Left Option > Select "Esc+" radio button works, but I didn't need to do that.

**Comment by paurana** [comment]
Thank you for this! Super helpful :)

**Comment by arwiz-077** [comment]
The skhd config is not working for me :(
I copied it like it is up there does someone know what the problem could be?

**Reply by padi** [reply]
If you're using iTerm2, try disabling iTerm2 > "Secure Keyboard Entry" and report back if it works for you.

**Comment by msanjayachar** [comment]
alt = option key

**Comment by msanjayachar** [comment]
Everything from the config is working for me.
Except:
1. Moving window to prev and next space
2. Moving window to a specific space with number

Anyone facing the same issue?
